src/main/scala/io/iohk/ethereum/mining/BlockGenerator.scala

```diff
@@ -75,6 +73,22 @@ class BlockGenerator(blockchainStorages: BlockchainStorages, blockchainConfig: B
         }
       }.getOrElse(Left(NoParent))
     }
+
+    result.right.foreach(b => cache.updateAndGet(new UnaryOperator[List[Block]] {
+      override def apply(t: List[Block]): List[Block] =
+        (b :: t).take(cacheLimit)
+    }))
```

Comment:
Do we need this cache? We cannot produce a new block for mining until the recently mined block has been included, can we?